app/models/metric/purging.rb

```diff
@@ -85,9 +93,33 @@ def self.purge_hourly(older_than, window = nil, total_limit = nil, &block)
   end
 
   def self.purge_realtime(older_than, window = nil, total_limit = nil, &block)
-    purge_by_date(older_than, "realtime", window, total_limit, &block)
+    truncate_child_tables(older_than)
   end
 
+  # truncate metrics child tables
+  # Determines hours not being preserved and truncates them
+  # Used for realtime metrics.
+  def self.truncate_child_tables(older_than)
+    target_hours = determine_target_hours(older_than, Time.now.utc)
+    return if target_hours.blank?
+
+    target_hours.each do |hour|
+      Metric.connection.truncate(Metric.reindex_table_name(hour), "Metric Truncate table #{hour}")
```

Comment:
I like this.  👍 

I think a developer who didn't happen to read the comments might see this now and say "huh, what is `reindex_table_name`", and hopefully would be able to grok what is happening with this before trying to make a change.

Reply:
thnx